spec/ndx-pose.extensions.yaml

```diff
@@ -92,27 +121,87 @@ groups:
       dtype: text
       doc: Version string of the software tool used.
       required: false
-  - name: nodes
+  groups:
+  - neurodata_type_inc: PoseEstimationSeries
+    doc: Estimated position data for each body part.
+    quantity: '*'
+  links:
+  - target_type: Skeleton
+    doc: Layout of body part locations and connections.
+- neurodata_type_def: TrainingFrame
+  neurodata_type_inc: NWBDataInterface
+  default_name: TrainingFrame
+  doc: Group that holds ground-truth position data for all instances of a skeleton in a single frame.
+  attributes:
+  - name: annotator
     dtype: text
-    dims:
-    - num_body_parts
-    shape:
-    - null
-    doc: Array of body part names corresponding to the names of the PoseEstimationSeries
-      objects within this group.
+    doc: Name of annotator who labeled the TrainingFrame.
+    required: false
+  groups:
+  - neurodata_type_inc: Instance
+    doc: Position data for a single instance of a skeleton in a single training frame.
+    quantity: '*'
+  - name: source_video
+    neurodata_type_inc: ImageSeries
+    doc: Path to original video file and frame used.
     quantity: '?'
-  - name: edges
+    attributes:
+    - name: path
+      dtype: text
+      doc: Path to original video file.
+      required: false
+    - name: frame_index
+      dtype: uint8
+      doc: Frame index of TrainingFrame in original video file.
+      required: false
+  - name: source_frame
+    neurodata_type_inc: Image
+    doc: Image frame used for training (stored either internally or externally).
```

Comment:
Hi @CBroz1, @rly,

I believe all the changes have been address (except for `Image` only allowing externally stored images and our current docs incorrectly say "stored either internally or externally"). With `source_video` linked to an `ImageSeries`, I believe we may not even need the `source_frame` group as we can directly supply a collection of images. However, the images supplied for training are rarely in a sequential format (and may therefore be an incorrect use of `ImageSeries` which extends `TimeSeries`).

**The problem**: We would like to store non-sequential images internally. `ImageSeries` allows images to be stored internally, but expects them to be sequential. `Image` allows images to be non-sequential, but requires them to be stored externally.

@CodyCBakerPhD had suggested making a change to the `Image` datatype to allow this. Is this feasible? Is there another workaround? Should we just hack `ImageSeries` for our purposes?

Reply:
In NWB 2.5.0, we recently revamped the [`Images` neurodata type](https://github.com/NeurodataWithoutBorders/nwb-schema/blob/2.5.0/core/nwb.base.yaml#L245), which represents a collection of images that may or may not be ordered. The [`IndexSeries` neurodata type](https://github.com/NeurodataWithoutBorders/nwb-schema/blob/2.5.0/core/nwb.image.yaml#L178) supports references to an `Images` type. This type seems like a better fit for the use case. 

And to be clear, `Image` requires images to be stored internally and does not support external storage. External storage of individual images is not currently supported, but we can add that if we need to. External storage of movies is supported through `ImageSeries`.

Reply:
Ok, great. I swapped the Image data type for the Images data type and added an `images_index` attribute to pull out a specific frame from the `Images` collection.